app/javascript/gobierto_data/webapp/components/commons/Visualizations.vue

```diff
@@ -44,6 +65,37 @@ export default {
     getConfig() {
       // export the visualization configuration object
       return this.viewer.save()
+    },
+    enableDisabledPerspective(value) {
+      const shadowRootPerspective = document.querySelector('perspective-viewer').shadowRoot
+      const sidePanelPerspective = shadowRootPerspective.getElementById('side_panel')
+      const topPanelPerspective = shadowRootPerspective.getElementById('top_panel')
+      topPanelPerspective.style.display = value
+      sidePanelPerspective.style.display = value
+    },
+    listenerPerspective() {
+      const shadowRootPerspective = document.querySelector('perspective-viewer').shadowRoot
+      const configButtonPerspective = shadowRootPerspective.getElementById('config_button')
+      configButtonPerspective.style.display = "none"
+      const selectVizPerspective = shadowRootPerspective.getElementById('vis_selector')
+      const showDialog = this.showSavingDialog
+      const sendSelectedValue = this.selectedValue
+      let selectedValue
+
+      selectVizPerspective.addEventListener('change', function() {
+        selectedValue = selectVizPerspective.options[selectVizPerspective.selectedIndex].value;
+        showDialog()
+        sendSelectedValue(selectedValue)
+      })
+    },
+    showSavingDialog() {
+      this.$emit("showSaving")
+    },
+    selectedValue(chart) {
+      this.$emit("selectedChart", chart)
+    },
```

Comment:
1. `showDialog`: you are creating a variable, to store a function to emit later an event. Nonsense, quite hard to follow/understand.
2. `selectedValue`: you create a variable uninitializated, another to store a function with the same name as the first variable; then set the value in the event to emit it in the function. WTF.

Just emit both events inside the listener. Simple as that.

```suggestion
      selectVizPerspective.addEventListener('change', () => {
        selectedValue = selectVizPerspective.options[selectVizPerspective.selectedIndex].value;
        
        this.$emit("showSaving")
        this.$emit("selectedChart", selectedValue)
      })
    },
```

If you need to keep the `this` context, just use an arrow function.

When the `change` event is triggered? Why do we need two events to handle this?